server/test/unit/server/controllers/test_content.py

```diff
@@ -157,13 +158,25 @@ class TestDownloadRepo(TestCase):
     @patch(MODULE_PATH + 'LazyUnitDownloadStep')
     @patch(MODULE_PATH + '_create_download_requests')
     @patch(MODULE_PATH + 'repo_controller.find_units_not_downloaded')
-    def test_download_deferred(self, mock_missing_units, mock_create_requests, mock_step):
-        """Assert the download step is initialized and called."""
+    def test_download_repo_no_verify(self, mock_missing_units, mock_create_requests, mock_step):
+        """Assert the download step is initialized and called with missing units."""
         content_controller.download_repo('fake-id')
         mock_missing_units.assert_called_once_with('fake-id')
         mock_create_requests.assert_called_once_with(mock_missing_units.return_value)
         mock_step.return_value.process_lifecycle.assert_called_once_with()
 
+    @patch(MODULE_PATH + 'itertools.chain')
```

Comment:
As you've written it, this is definitely the most concise way to do the test, which is valuable. Mocking ``chain`` raised an eyebrow though. I think tests are more valuable when they can make assertions that desired behaviors happened, not that specific implementations were executed. In a case like this, I prefer to assign a more representative return value to ``mock_units_qs``, and then assert that the expected transformation of that was passed into ``mock_create_requests``.

More concretely, something along these lines:

```
qs1 = mock.Mock()
qs2 = mock.Mock()
mock_units_qs.return_value = [[qs1], [qs2]]
...
self.assertEqual(list(mock_create_requests.call_args[0][0]), [qs1, qs2])
```

It's a small difference especially in this case, but the closer we can stick to letting representative data flow through our unit tests, the more valuable those tests will be.